src/asvz_bot.py

```diff
@@ -197,6 +197,17 @@ def from_lesson_attributes(
                 By.XPATH, ".//a[starts-with(@href, '{}')]".format(LESSON_BASE_URL)
             ).get_attribute("href")
             logging.debug(f"Found lesson url: {lesson_url}")
+
+            #When there is no lesson on the requested day, the ASVZ webpage returns the first lesson on the next day with lessons.
+            driver.get(lesson_url)
+            driver.implicitly_wait(3)
+            lesson_start = AsvzEnroller.__get_enrollment_and_start_time(driver)[1]
+            if lesson_start.weekday() != weekday_int:
```

Comment:
Id prefer no to compare it to the weekday only, as there may be a whole week of holiday and then it might match the same lesson a week later. What do you think?

Reply:
Good observation! I think that indeed makes more sense and follows the semantics of the program more closely.